rfcs/NNNN-decorator-roadmap.md

```diff
@@ -0,0 +1,217 @@
+- TypeScript developers not yet using decorators because of forward compatibility concerns
+- TypeScript developers stuck using legacy experimental decorators because of backward compatibility concerns
+
+This plan needs to address all audiences during the migration.
+
+### End State
+
+After all stages of the implementation plan are complete:
+
+- Standard decorators are the only decorator implementation
+- Decorators are the one way to define reactive properties
+- Remove `static properties`, `static createProperty()` and associated APIs
+- Remove `static addInitializer()` since the standard decorator API provides this
+- Core decorators (`@property`, `@state`) are exported from the main `reactive-element`, `lit-element` and `lit` modules.
+
+### Standard Decorators
+
+Standard decorators have a much different design than TypeScript's experimental decorators. Rather than getting access to the class and making dynamic, imperative changes to the class, standard decorators have no direct access to the class or the class member they're decorating and must instead return an object describing a replacement for the class member. For instance class accessor decorators return an object with get and set methods.
+
+This makes standard decorator usage must more static than experimental decorators. They can't add new class members, and can't change the type of member they decorate, and ReactiveElement's `static createProperty()` API is simply not callable.
+
+The good news is that our decorator implementations are likely to be smaller and simpler. For example, the two main functions of our `@property()` decorator are to replace a field with a getter/setter pair that calls `requestUpdate()` in the setter, and to store metadata about the field for use in attribute reflection.
+
+For example, a simplified decorator that only replaces an accessor might look like:
+
+```ts
+export const property =
+  (options?: PropertyDeclaration) =>
+  <C extends ReactiveElement, V>(
+    _target: ClassAccessorDecoratorTarget<C, V>,
+    context: ClassAccessorDecoratorContext<C, V>
+  ): ClassAccessorDecoratorResult<C, V> => {
+    const {access, name} = context;
+    return {
+      get(this: C) {
+        return access.get(this);
+      },
+      set(this: C, v: V) {
+        const oldValue = access.get(this);
+        access.set(this, v);
+        this.requestUpdate(name, oldValue);
+      },
+    };
+  };
+```
+
+This decorator would require the new "auto accessor" feature and `accessor` keyword to use:
+
+```ts
+class MyElement extends LitElement {
+  @property()
+  accessor foo = 42;
+}
+```
+
+## Implementation Considerations
+
+### Implementation Plan
+
+The plan is divided into stages which must be shipped as releases.
+
+#### I. Add standard decorators (non-breaking)
+
+This stage adds support for the new decorators standard in a backward-compatible way for both TypeScript and JavaScript developers.
+
+##### Requirements
+* TypeScript ships decorator metadata
```

Comment:
Worth mentioning that this is ETA TypeScript 5.2, slated for release at the end of August: https://github.com/microsoft/TypeScript/issues/54298